keras_cv/models/generative/stable_diffusion/clip_tokenizer.py

```diff
@@ -89,18 +89,55 @@ def __init__(self, bpe_path=None):
         for merge in merges:
             vocab.append("".join(merge))
         vocab.extend(["<|startoftext|>", "<|endoftext|>"])
+        self.vocab = vocab
         self.encoder = dict(zip(vocab, range(len(vocab))))
         self.decoder = {v: k for k, v in self.encoder.items()}
         self.bpe_ranks = dict(zip(merges, range(len(merges))))
+
+        self.special_tokens = {
+            "<|startoftext|>": "<|startoftext|>",
+            "<|endoftext|>": "<|endoftext|>",
+        }
         self.cache = {
             "<|startoftext|>": "<|startoftext|>",
             "<|endoftext|>": "<|endoftext|>",
         }
-        self.pat = re.compile(
-            r"""<\|startoftext\|>|<\|endoftext\|>|'s|'t|'re|'ve|'m|'ll|'d|[\p{L}]+|[\p{N}]|[^\s\p{L}\p{N}]+""",
+        self.pat = self._create_pat()
+        self._overridden = False
+
+    def _create_pat(self):
+        return re.compile(
+            "|".join([re.escape(key) for key in self.special_tokens.keys()])
+            + r"""|'s|'t|'re|'ve|'m|'ll|'d|[\p{L}]+|[\p{N}]|[^\s\p{L}\p{N}]+""",
             re.IGNORECASE,
         )
 
+    @property
+    def end_of_text(self):
+        return self.encoder["<|endoftext|>"]
+
+    @property
+    def start_of_text(self):
+        return self.encoder["<|startoftext|>"]
+
+    def add_tokens(self, *args):
+        token = args[0]
+        if self._overridden:
+            raise ValueError("For now KerasCV can only finetune on one special token.")
+        self._overridden = True
+        self.vocab[48973] = token
+        # 48973 is the index of ðŁĺĤðŁĺĤðŁĺĤðŁĺĤðŁĺĤðŁĺĤðŁĺĤ</w>"
```

Comment:
What is the usecase for this token? If it is for classification loss like BERT the established practice is to use `0`.

Reply:
Unfortunately I just looked through CLIP and there are no reserved tokens like that.

That is really unfortunate; but there ARE many garbage tokens we could throw out and replace